### ISSUE TITLE
Many professions fail to spawn nested contents

### ISSUE THREAD

**Issue by esotericist** [CleverRaven/Cataclysm-DDA#41969]
# Describe the bug

many professions specify nested contents in the form of 
```
          { "item": "medium_plus_battery_cell", "ammo-item": "battery", "charges": 600, "container-item": "mil_mess_kit" },
```
and
```
          {
            "item": "m2010",
            "ammo-item": "300_winmag",
            "charges": 5,
            "contents-item": [ "shoulder_strap", "rifle_scope", "bipod" ]
          },
```

which apparently fails to insert into those items due to pocket type mismatches?

# Steps To Reproduce

1. apply wapcaplet's small patch here for debug data: https://gist.github.com/wapcaplet/7ca7dccdd97d28d75b3462859d8750d8
2. start character creation
3. cycle through professions
4. watch the terminal output show `put_in failed` messages
5. notice how the containers (weapons, tools, clothing) spawn but not what goes in them (weapon mods, battery cells, etc)

# Expected behavior

stuff specified in professions should get their contents.

presumably item group json should either be more permissive when inserting things, or updated to use the correct pocket type, or something?

# Screenshots

![image](https://user-images.githubusercontent.com/1569754/87022865-ec761d00-c18b-11ea-8001-7cc2b5b20951.png)
![image](https://user-images.githubusercontent.com/1569754/87022911-fac43900-c18b-11ea-85f4-4ae3d62d46f1.png)

# Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 19.10; Release: 19.10; Codename: eoan; 
- Game Version: 0.E-3911-g312db57cf0c0d-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Magiclysm [magiclysm]
]

# Additional context

since this seems to be an item groups issue, it very likely affects actual loot items out in the wild instead of just being limited professions, i'm just not in a position to verify that

other things also fail to go into containers, but not for the above reasons; for instance, M24 fails to insert in back holster on the police sniper weapon, but that's because the M24 is too long for the back holster to hold. so not all profession put_in errors are this problem.

**Comment by misterprimus** [comment]
I currently have issues spawning ammo or mags in spawned guns of downed zeds (or even ground-spawned guns) so I think you're right.

**Comment by wapcaplet** [comment]
*Note*, the patch mentioned in the issue description is no longer necessary since #41677 added an explicit `debugmsg` to report `put_in` failures here.

I have checked the full list of professions in current master 963edfda and none of them are reporting failures when scrolling through the starting profession list. I checked one profession (Soldier) into an actual game and saw the M4A1 is correctly spawned with shoulder strap and holographic sight, and I believe all the others should be working now as well.

There is still a discrepancy, that `can_contain` in fact returns different results than `insert_item` for gunmods. For example `item( "m4a1" ).contents.can_contain( item( "shoulder_strap" ) )` will fail, while `insert_item` with the same gun/gunmod succeeds. I will submit a PR with test cases for this, but it does not appear to have an adverse effect on starting profession spawns.

**Comment by LaVeyanFiend** [comment]
![image](https://user-images.githubusercontent.com/51099123/93026614-499bae00-f5d5-11ea-9752-801486702af4.png)
Confirmed with Military Recruit

**Comment by mqrause** [comment]
> _Note_, the patch mentioned in the issue description is no longer necessary since #41677 added an explicit `debugmsg` to report `put_in` failures here.

Note to the note: I kind of want to revert (or rework) that in #44155 because it fires even when not appropriate. This is not the place to discuss it, I just wanted to mention it here in case things change because of that.

**Comment by akirashirosawa** [comment]
This issue was fixed in #41677 which the person above wants to revert (LOL).

Bionic Sniper with `300_winmag` ammo and `"shoulder_strap", "rifle_scope", "bipod"` mods:
<img width="541" alt="assasin" src="https://user-images.githubusercontent.com/38557723/93065400-5970cc00-f681-11ea-9d5f-0b2ad98d2ffc.png">

You can close this issue.
